java/read-symbols-mobile-style-file/src/main/java/com/esri/arcgisruntime/sample/readsymbolsmobilestylefile/MainActivity.java

```diff
@@ -0,0 +1,449 @@
+
+  private RecyclerView mEyesRecyclerView;
+  private RecyclerView mMouthRecyclerView;
+  private RecyclerView mHatRecyclerView;
+  private ImageView mPreviewView;
+  private Spinner mColorSpinner;
+  private SymbolAdapter mEyesAdapter;
+  private SymbolAdapter mMouthAdapter;
+  private SymbolAdapter mHatAdapter;
+
+  private final Map<String, SymbolStyleSearchResult> mSelectedSymbols = new HashMap<>();
+  private String mFaceSymbolKey;
+  private ArrayList<String> mKeys = new ArrayList<>();
+  private int mColor = -1;
+  private int mSize = 25;
+
+  private MapView mMapView;
+  private GraphicsOverlay mGraphicsOverlay;
+  private SymbolStyle mEmojiStyle;
+  private MultilayerPointSymbol mCurrentMultilayerSymbol;
+
+  @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    mMapView = findViewById(R.id.mapView);
+    mEyesRecyclerView = findViewById(R.id.eyesRecyclerView);
+    mMouthRecyclerView = findViewById(R.id.mouthRecyclerView);
+    mHatRecyclerView = findViewById(R.id.hatRecyclerView);
+    mPreviewView = findViewById(R.id.previewView);
+
+    // create a map
+    ArcGISMap map = new ArcGISMap(Basemap.createTopographic());
+    // add the map to the map view
+    mMapView.setMap(map);
+
+    // create a graphics overlay to add graphics to and add it to the map view
+    mGraphicsOverlay = new GraphicsOverlay();
+    mMapView.getGraphicsOverlays().add(mGraphicsOverlay);
+
+    // add listener to handle motion events when the user taps on the map view
+    mMapView.setOnTouchListener(
+        new DefaultMapViewOnTouchListener(this, mMapView) {
+          @Override
+          public boolean onSingleTapConfirmed(MotionEvent motionEvent) {
+            addGraphic(mGraphicsOverlay, mapPointFrom(mMapView, motionEvent), mCurrentMultilayerSymbol);
+            return true;
+          }
+        });
+
+    // add a seek bar to change the size of the current multilayer symbol
+    SeekBar sizeSeekBar = findViewById(R.id.sizeSeekBar);
+    // set initial progress to 25
+    sizeSeekBar.setProgress(mSize);
+    sizeSeekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
+      @Override public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
+        setSymbolSize(progress);
+      }
+
+      @Override public void onStartTrackingTouch(SeekBar seekBar) {
+
+      }
+
+      @Override public void onStopTrackingTouch(SeekBar seekBar) {
+
+      }
+    });
+
+    // add a spinner to change the color of the first layer of the multilayer symbol
+    mColorSpinner = findViewById(R.id.colorSpinner);
+    mColorSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+      @Override public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        setLayerColor(position);
```

Comment:
There's a crash on first launch when user hasn't allowed read permission:

```
java.lang.NullPointerException: Attempt to invoke virtual method 'com.esri.arcgisruntime.concurrent.ListenableFuture com.esri.arcgisruntime.symbology.SymbolStyle.getSymbolAsync(java.util.List)' on a null object reference
        at com.esri.arcgisruntime.sample.readsymbolsmobilestylefile.MainActivity.createSwatchAsync(MainActivity.java:224)
        at com.esri.arcgisruntime.sample.readsymbolsmobilestylefile.MainActivity.setLayerColor(MainActivity.java:307)
        at com.esri.arcgisruntime.sample.readsymbolsmobilestylefile.MainActivity.access$500(MainActivity.java:68)
        at com.esri.arcgisruntime.sample.readsymbolsmobilestylefile.MainActivity$3.onItemSelected(MainActivity.java:145)
        at android.widget.AdapterView.fireOnSelected(AdapterView.java:944)
```

This is because the spinner is inflated and selectes the first item. It might be worth removing the `Color` label above the spinner and adding a `Color...` item to the spinner at index 0. Then you can check if the selected index is 0 before calling `setLayerColor`. Or only populating the spinner after everything has loaded.